src/sleep/sleep.c

```diff
@@ -617,7 +617,9 @@ static int run(int argc, char *argv[]) {
                 if (r < 0)
                         return r;
 
-                (void) unit_freezer_freeze(user_slice_freezer);
+                r = unit_freezer_freeze(user_slice_freezer);
+                if (r < 0)
+                        (void) unit_freezer_thaw(user_slice_freezer);
```

Comment:
same here, we should really do that server side, not client side: if we determine server side that a freeze doesn't work, then thaw immediately server side.

Reply:
resolving as per https://github.com/systemd/systemd/pull/36778#discussion_r2008756351